src/containers/ClassifierContainer.jsx

```diff
@@ -168,11 +169,24 @@ class ClassifierContainer extends React.Component {
       </main>
     );
   }
+  
+  //----------------------------------------------------------------
+  
+  componentWillReceiveProps(next) {
+    if (!this.props.selectedAnnotation && next.selectedAnnotation) {
+      this.setState({
+        popup: <SelectedAnnotation annotation={next.selectedAnnotation} />
+      });
+    }
+  }
 
   //----------------------------------------------------------------
 
   closePopup() {
     this.setState({ popup: null });
```

Comment:
The whole `containers/ClassifierContainer.jsx` was a mistake (the correct one is `components/ClassifierContainer.jsx`) so this will be removed altogether.

However, the point you made is still pertinent: should the PopUp state be in _each individual_ component?

If we have a sufficiently large number of different components (e.g. Classifier Page, hypothetical Editor Page, hypothetical Document Browser) that each use the Popups differently, then yes, we need individual popup states in each components.

Otherwise, no, a global popup would work just as fine. (Just be aware of how the onClose/etc functions hook up between components, and remember to `.bind()` everything properly.)

I like compartmentalised (one-per-component) popups personally, but I find it hard to find a logical argument against a single global popup state which looks so much cleaner.